### ISSUE TITLE
download files from 2x nested data set lists

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/galaxy#5186]
I'f following the GTN tutorial for mothurs 454 SOP. 

In step 4: Visualisations make.biom creates a (2x) nested data set list containing a single biom file. When I try to download the file nothing happens, but  the following appears in the logs: 

```
141.65.26.175 - - [11/Dez/2017:13:14:02 +0200] "GET /datasets/78f26c71af49d8d0/display?to_ext=biom1&hdca_
id=6014235b5615124b&element_identifier=0.03 HTTP/1.1" 500 - "https://galaxy.intranet.ufz.de/" "Mozilla/5.
0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0"
Error - <type 'exceptions.AttributeError'>: 'NoneType' object has no attribute 'history'
URL: http://galaxy.intranet.ufz.de/datasets/78f26c71af49d8d0/display?to_ext=biom1&hdca_id=6014235b5615124
b&element_identifier=0.03
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/web/framework/middleware/error.py', line 154 in __call_
_
  app_iter = self.application(environ, sr_checker)
File '/gpfs1/data/galaxy_server/galaxy/.venv/lib/python2.7/site-packages/paste/recursive.py', line 85 in 
__call__
  return self.application(environ, start_response)
File '/gpfs1/data/galaxy_server/galaxy/.venv/lib/python2.7/site-packages/paste/httpexceptions.py', line 6
40 in __call__
  return self.application(environ, start_response)
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/web/framework/base.py', line 136 in __call__
  return self.handle_request(environ, start_response)
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/web/framework/base.py', line 215 in handle_request
  body = method(trans, **kwargs)
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/webapps/galaxy/controllers/dataset.py', line 231 in dis
play
  hdca = self.app.dataset_collections_service.get_dataset_collection_instance(trans, "history", hdca_id)
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/managers/collections.py', line 317 in get_dataset_collection_instance
  return self.__get_history_collection_instance(trans, id, **kwds)
File '/gpfs1/data/galaxy_server/galaxy/lib/galaxy/managers/collections.py', line 332 in __get_history_collection_instance
  self.history_manager.error_unless_accessible(collection_instance.history, trans.user, current_history=trans.history)
AttributeError: 'NoneType' object has no attribute 'history'


CGI Variables
-------------
  CONTENT_LENGTH: '0'
  HTTP_ACCEPT: 'text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8'
  HTTP_ACCEPT_ENCODING: 'gzip, deflate, br'
  HTTP_ACCEPT_LANGUAGE: 'en-US,en;q=0.5'
  HTTP_CONNECTION: 'Keep-Alive'
  HTTP_COOKIE: 'galaxysession=79a1fb9ed4bcf577a9aa4b882d76880d48405f091dfe61c3935edfb2bc302a2b70a9c6abbae367ba'
  HTTP_DNT: '1'
  HTTP_HOST: 'galaxy.intranet.ufz.de'
  HTTP_REFERER: 'https://galaxy.intranet.ufz.de/'
  HTTP_UPGRADE_INSECURE_REQUESTS: '1'
  HTTP_USER_AGENT: 'Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:57.0) Gecko/20100101 Firefox/57.0'
  HTTP_X_FORWARDED_FOR: '141.65.26.175'
  HTTP_X_FORWARDED_HOST: 'galaxy.intranet.ufz.de'
  HTTP_X_FORWARDED_SERVER: 'galaxy.intranet.ufz.de'
  ORGINAL_HTTP_HOST: 'localhost:8080'
  ORGINAL_REMOTE_ADDR: '127.0.0.1'
  PATH_INFO: '/datasets/78f26c71af49d8d0/display'
  QUERY_STRING: 'to_ext=biom1&hdca_id=6014235b5615124b&element_identifier=0.03'
  REMOTE_ADDR: '141.65.26.175'
  REQUEST_METHOD: 'GET'
  SERVER_NAME: '0.0.0.0'
  SERVER_PORT: '8080'
  SERVER_PROTOCOL: 'HTTP/1.1'


WSGI Variables
--------------
  application: <paste.recursive.RecursiveMiddleware object at 0x7fb879a386d0>
  controller_action_key: u'web.dataset.display'
  is_api_request: False
  paste.cookies: (<SimpleCookie: galaxysession='79a1fb9ed4bcf577a9aa4b882d76880d48405f091dfe61c3935edfb2bc302a2b70a9c6abbae367ba'>, 'galaxysession=79a1fb9ed4bcf577a9aa4b882d76880d48405f091dfe61c3935edfb2bc302a2b70a9c6abbae367ba')
  paste.expected_exceptions: [<class 'paste.httpexceptions.HTTPException'>]
  paste.httpexceptions: <paste.httpexceptions.HTTPExceptionHandler object at 0x7fb879ac3610>
  paste.httpserver.proxy.host: 'dummy'
  paste.httpserver.proxy.scheme: 'http'
  paste.httpserver.thread_pool: <paste.httpserver.ThreadPool object at 0x7fb855783610>
  paste.parsed_querystring: ([('to_ext', 'biom1'), ('hdca_id', '6014235b5615124b'), ('element_identifier', '0.03')], 'to_ext=biom1&hdca_id=6014235b5615124b&element_identifier=0.03')
  paste.recursive.forward: <paste.recursive.Forwarder from />
  paste.recursive.include: <paste.recursive.Includer from />
  paste.recursive.include_app_iter: <paste.recursive.IncluderAppIter from />
  paste.recursive.script_name: ''
  paste.throw_errors: True
  request_id: 'c6805ee4de6c11e79e7f1866daebcee4'
  webob._parsed_query_vars: (GET([(u'to_ext', u'biom1'), (u'hdca_id', u'6014235b5615124b'), (u'element_identifier', u'0.03')]), 'to_ext=biom1&hdca_id=6014235b5615124b&element_identifier=0.03')
  wsgi process: 'Multithreaded'
------------------------------------------------------------
```

When I remove one level of nesting with the tool `Flatten Collection` the download works. 

So its likely a problem with 2x nesting... 

I'm working on Galaxy 17.09

**Comment by jmchilton** [comment]
This will be fixed with the release of 18.01 - the fix is from @mvdbeek in https://github.com/galaxyproject/galaxy/pull/4934.

**Comment by bernt-matthias** [comment]
Great. Thanks for the info.
